mission_control/navigator_missions/nav_missions/pinger.py

```diff
@@ -49,10 +54,14 @@ def get_objects(self):
 
         self.gate_poses = np.array([gate_1_pos, gate_2_pos, gate_3_pos])
 
+    @txros.util.cancellableInlineCallbacks
     def get_observation_poses(self):
         """Set 2 points to observe the pinger from, in front of gates 1 and 3"""
         self.get_gate_perp()
-        pose = yield self.navigator.tx_pose[:2]
+        #Make sure they are actually in a line
+        if np.isnan(self.g_perp[0]) or np.isnan(self.g_perp[1]):
+            raise Exception("Gates are not in a line")
+        pose = (yield self.navigator.tx_pose)[0][:2]
```

Comment:
If this is the only thing keeping this function from being synchronous? There's a secret `navigator.pose` that gets updated on each odom message so you can use that instead of decorating this function.

I feel like the less defer functions the cleaner the code.